src/site/antora/modules/ROOT/pages/manual/configuration.adoc

```diff
@@ -14,1856 +14,1019 @@
+===== Global configuration attributes
+
+The XML format supports the following additional attributes on the `Configuration` element.
+
+[id=configuration-attribute-schema]
+====== `schema`
+
+[cols="1h,5"]
+|===
+| Type          | classpath resource
+| Default value | `null`
+|===
 
-[#XInclude]
-== XInclude
+Specifies the path to a classpath resource containing an XML schema.
+
+[id=configuration-attribute-strict]
+====== `strict`
+
+[cols="1h,5"]
+|===
+| Type          | `boolean`
+| Default value | `false`
+|===
+
+If set to `true` all configuration files will be checked against the XML schema provided by the
+<<configuration-attribute-schema>>.
+
+This setting also enables "XML strict mode" and allows to specify the **plugin type** of an element through a `type` attribute instead of the tag name.
+
+[id=xinclude]
+===== [[XInlcude]] XInclude
 
 XML configuration files can include other files with
-http://www.xml.com/lpt/a/1009[XInclude]. Here is an example log4j2.xml
-file that includes two other files:
+https://www.w3.org/TR/xinclude/[XInclude].
+
+NOTE: The list of `XInclude` and `XPath` features supported depends upon your
+https://docs.oracle.com/javase/{java-target-version}/docs/technotes/guides/xml/jaxp/index.html[JAXP implementation].
+
+Here is an example log4j2.xml file that includes two other files:
 
 .log4j2.xml
 [source,xml]
 ----
-<?xml version="1.0" encoding="UTF-8"?>
-<configuration xmlns:xi="http://www.w3.org/2001/XInclude"
-               status="warn" name="XIncludeDemo">
-  <properties>
-    <property name="filename">xinclude-demo.log</property>
-  </properties>
-  <ThresholdFilter level="debug"/>
-  <xi:include href="log4j-xinclude-appenders.xml" />
-  <xi:include href="log4j-xinclude-loggers.xml" />
-</configuration>
+include::example$configuration/xinclude-main.xml[lines=1;18..-1]
 ----
 
-.log4j-xinclude-appenders.xml
+.xinclude-appenders.xml
 [source,xml]
 ----
-<?xml version="1.0" encoding="UTF-8"?>
-<appenders>
-  <Console name="STDOUT">
-    <PatternLayout pattern="%m%n" />
-  </Console>
-  <File name="File" fileName="${filename}" bufferedIO="true" immediateFlush="true">
-    <PatternLayout>
-      <pattern>%d %p %C{1.} [%t] %m%n</pattern>
-    </PatternLayout>
-  </File>
-</appenders>
+include::example$configuration/xinclude-appenders.xml[lines=1;18..-1]
 ----
 
-.log4j-xinclude-loggers.xml
+.xinclude-loggers.xml
 [source,xml]
 ----
-<?xml version="1.0" encoding="UTF-8"?>
-<loggers>
-  <logger name="org.apache.logging.log4j.test1" level="debug" additivity="false">
-    <ThreadContextMapFilter>
-      <KeyValuePair key="test" value="123" />
-    </ThreadContextMapFilter>
-    <AppenderRef ref="STDOUT" />
-  </logger>
+include::example$configuration/xinclude-loggers.xml[lines=1;18..-1]
+----
 
-  <logger name="org.apache.logging.log4j.test2" level="debug" additivity="false">
-    <AppenderRef ref="File" />
-  </logger>
+[id=java-properties-features]
+==== Java properties format
```

Comment:
Shall we add a `TIP` here and advice users to migrate to XML, etc. (with some brief motivation) if possible?

Reply:
We could add:
> Here is a picture of one of the authors, after converting several configuration files in Java properties format: https://www.deviantart.com/brown73/art/GORILLA-MAD-223018595
>
> If you don't want to end up like him, use to another format like XML or YAML.

Reply:
I added a tip in be54ca2aefd708b8159548277d99e12aea948980.